include/nuttx/arch.h

```diff
@@ -2555,8 +2555,10 @@ void arch_sporadic_resume(FAR struct tcb_s *tcb);
  *
  *   The second interface simple converts an elapsed time into well known
  *   units.
+ *
  ****************************************************************************/
 
+void up_perf_init(FAR void *arg);
```

Comment:
```suggestion
void up_perf_init(uintptr_t arg);
```
Just an option to have less type casts.
I'm fine with current way.
Or we can make it more explicit `void up_perf_init(uint32_t cpu_freq);`

Reply:
Just feedback if you will apply suggestion, so I can either wait for update or merge it as is.

Reply:
It intend to use void * here, if other arch implementation may need more argument, he can define a struct to contain more info without modify the prototype.

Reply:
`uintptr_t` will give same effect. But let's keep `void *`